starter-code/scripts/models/article.js

```diff
@@ -69,18 +69,21 @@
     // NOTE: Grab the word count from each article body.
       return article.body.split(' ').length;
     })
-    // TODO: complete this reduce to get a grand total word count
-    .reduce(function() {
+    // TODONE: complete this reduce to get a grand total word count
+    .reduce(function(a, b) {
+      return a + b;
     });
   };
 
   /* Chain together a `map` and a `reduce` call to
           produce an array of *unique* author names. */
   Article.allAuthors = function() {
-  //return       TODO: map our collection
-    //return    TODO: return just the author names
-
-  /* TODO: For our reduce that we'll chain here -- since we are trying to
+    return Article.allArticles.map(function(curArticle) {
+      return curArticle.author;
+    }).filter(function(author, index, array){
+      return array.indexOf(author) === index;
+    });
+  /* TODONE: For our reduce that we'll chain here -- since we are trying to
```

Comment:
Great job! I hope the lecture on reduce() was helpful!. Rick demonstrated one way of using it here. When I completed this assignment, here's what I did:
```
.reduce(function(acc, cur){
      if (!acc.includes(cur)){
        acc.push(cur);
      }
      return acc;
    }, []);
```